src/client/Microsoft.Identity.Client/AppConfig/CiamAuthorityHelper.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    internal class CiamAuthorityHelper
+    {
+        public CiamAuthorityHelper(Uri authority)
+        {
+            TransformAuthority(authority);
+        }
+
+        public CiamAuthorityHelper(string instance, string tenant)
+        {
+            instance = instance.Trim().TrimEnd('/');
+            if (string.IsNullOrEmpty(tenant))
+            {
+                TransformAuthority(new Uri(instance + '/' + tenant));
+            }
+            else
+            {
+                TransformedInstance = instance;
+                TransformedTenant = tenant;
+                TransformedAuthority = new Uri(instance + '/' + tenant);
+            }
+        }
+
+        public Uri TransformedAuthority { get; private set; }
+
+        public string TransformedTenant { get; private set; }
+
+        public string TransformedInstance { get; private set; }
+
+        
+
+        private void TransformAuthority(Uri ciamAuthority)
+        {
+            string host = ciamAuthority.Host + ciamAuthority.AbsolutePath;
+            if (string.Equals(ciamAuthority.AbsolutePath, "/"))
+            {
+                string ciamTenant = host.Substring(0, host.IndexOf(".ciamlogin.com", StringComparison.OrdinalIgnoreCase));
+                TransformedInstance = "https://login.ciamlogin.com/";
```

Comment:
@trwalke: this should not be `"https://login.ciamlogin.com/"` any longer (this was in the initial prototype before the endpoint is available), but it should be `$"https://{ciamTenant}.ciamlogin.com/"` now.